internal/buildjob/buildjob.go

```diff
@@ -155,7 +155,11 @@ func (j *Job) Run() error {
 	return j.transitionToComplete(cib, images)
 }
 
-func (j *Job) Cleanup() {
+func (j *Job) Cleanup(forced bool) {
+	if forced {
+		j.log.Info("Caught kill signal, cleaning up")
+	}
+
 	for _, fn := range j.cleanupSteps {
 		fn()
 	}
```

Comment:
This type/func doesn't really know about signal trapping, right? Doesn't it make more sense to keep this signature as is and then log the "caught kill signal..." message inside the build command?

```go
stopper := make(chan os.Signal)
signal.Notify(stopper, os.Interrupt, syscall.SIGTERM)

job, err := buildjob.New(cfg)
if err != nil {
        panic(err)
}
defer job.Cleanup()

go func() {
	<-stopper
        log.Info("lorem ipsum...")
	job.Cleanup()
	os.Exit(0)
}()
```

Reply:
I didn't want to expose `job.Log` publicly since it needs the `annotations` to be caught by our internal fluentd, but I agree this is weird. I think we can do something better about the logging and remove this, let me think about it.

Reply:
Take a look at https://github.com/cerebrotech/domino/pull/20661 and the associated changes (e.g https://github.com/dominodatalab/forge/compare/steved/simplify-logging) and let me know if you think it's better?